src/main/java/com/newworld/saegil/simulator/stt/service/SpeechToTextService.java

```diff
@@ -0,0 +1,71 @@
+import java.util.Map;
+
+import org.springframework.ai.openai.OpenAiAudioTranscriptionModel;
+import org.springframework.ai.openai.OpenAiChatModel;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import lombok.RequiredArgsConstructor;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+@RequiredArgsConstructor
+@Service
+public class SpeechToTextService {
+
+    private final OpenAiAudioTranscriptionModel openAiAudioTranscriptionModel;
+    private final OpenAiChatModel chatModel;
+
+    public Mono<Map<String, String>> transcribeAndAnswer(MultipartFile audioFile) {
+        return Mono.fromCallable(() -> {
+                       try {
+                           InputStream audioStream = audioFile.getInputStream();
+                           Resource audioResource = new InputStreamResource(audioStream) {
+                               @Override
+                               public String getFilename() {
+                                   return audioFile.getOriginalFilename();
+                               }
+                           };
+                           return audioResource;
+                       } catch (IOException e) {
+                           throw new RuntimeException("Failed to process audio file", e);
+                       }
+                   })
+                   .flatMap(this::processAudioResource)
+                   .subscribeOn(Schedulers.boundedElastic());
+    }
+
+    public Mono<Map<String, String>> transcribeAndAnswer(byte[] audioFileBytes) {
+        return Mono.fromCallable(() -> {
+                       InputStream audioStream = new ByteArrayInputStream(audioFileBytes);
+                       Resource audioResource = new InputStreamResource(audioStream) {
+                           @Override
+                           public String getFilename() {
+                               return "audio.wav"; // Default filename
+                           }
+                       };
+                       return audioResource;
+                   })
+                   .flatMap(this::processAudioResource)
+                   .subscribeOn(Schedulers.boundedElastic());
+    }
+
+    private Mono<Map<String, String>> processAudioResource(Resource audioResource) {
+        return Mono.fromCallable(() -> openAiAudioTranscriptionModel.call(audioResource))
+                   .flatMap(transcribedText ->
+                           Mono.fromCallable(() -> chatModel.call(transcribedText))
+                               .map(answer -> Map.of(
+                                       "question", transcribedText,
+                                       "answer", answer
+                               ))
+                   )
+                   .subscribeOn(Schedulers.boundedElastic());
+    }
```

Comment:
함수형 호출이 많아서 가독성 있게 리팩토링해봤어요.

```suggestion
    public Mono<Map<String, String>> transcribeAndGetAnswer(final byte[] audioFileBytes) {
        return Mono.fromCallable(() -> generateConversation(audioFileBytes))
                   .subscribeOn(Schedulers.boundedElastic());
    }

    private Map<String, String> generateConversation(final byte[] audioFileBytes) {
        final Resource audioResource = convertToResource(audioFileBytes);
        final String transcribedText = openAiAudioTranscriptionModel.call(audioResource);
        final String answer = chatModel.call(transcribedText);

        return Map.of(
                "question", transcribedText,
                "answer", answer
        );
    }

    private Resource convertToResource(final byte[] audioFileBytes) {
        final InputStream audioStream = new ByteArrayInputStream(audioFileBytes);
        
        return new InputStreamResource(audioStream) {
            @Override
            public String getFilename() {
                return "audio.wav";
            }
        };
    }
```